java/src/main/java/com/google/cloud/dataproc/templates/dataplex/DataplexGCStoBQ.java

```diff
@@ -205,6 +207,24 @@ public static CommandLine parseArguments(String[] args) {
     }
   }
 
+  /**
+   * Execute request on Google API to fetch schema of a Dataplex entity and parses out a list of
+   * partition Keys
+   *
+   * @param entity name
+   * @return list with partition keys of the entity, return null if Dataplex entity has no
+   *     partitions
+   * @throws IOException when request on Dataplex API fails
+   */
+  private List<String> getPartitionKeyList(String entity) throws IOException {
+    try {
```

Comment:
Could you please provide more insight on why this is not desirable? To me it makes sense here, it is used to detect an edge case where source data has no partitions and allows for the template to adjust behavior accordingly.

I am more than open to change de design here but would like to understand why and also would love to hear suggestions on better approaches.

Reply:
I did not realized you have declared your own specific exception and you are catching that.
Often programmers catch generic exceptions which we want to avoid.
resolving it.